lib/contentful/client.rb

```diff
@@ -296,6 +298,16 @@ def sync(options = { initial: true })
 
     private
 
+    def ensure_select_is_doable(query)
+      return unless query.key?(:select)
+
+      query[:select] = query[:select].split(',').map(&:strip) if query[:select].is_a? String
+
+      query[:select] = query[:select].reject { |p| p.start_with?('sys.') }
```

Comment:
Curious why we're not allowing sys selections, which the docs even give as an explicit example?

https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/select-operator

My assumption is that the ruby library requires parts of the sys object for reasons?

If so, what if we just looked at the sys fields and ensured the sys fields we need are included. Otherwise, we could just allow individual sys selects to go through as is as long as we have what we need.

On the one hand I don't have a strong opinion about this, but on the other I think it's worth pushing on a bit to see if there's a reason we *must* return the whole sys object. The entire purpose of the `select` operator is to limit the amount of data returned, so we should avoid working against that functionality if it's at all possible.

I do think a comment would be worthwhile here explaining the reasoning if any for including the whole sys object and disallowing attribute selection in there, if we do have a good reason.